ui/app/src/state/epics/auth.ts

```diff
@@ -44,10 +44,19 @@ const epics: CrafterCMSEpic[] = [
     ),
   // endregion
   // region loginComplete
-  (action$) =>
+  (action$, state$) =>
     action$.pipe(
       ofType(loginComplete.type),
-      map(() => messageSharedWorker(refreshAuthToken()))
+      withLatestFrom(state$),
+      switchMap(([, state]) => {
+        const hasActiveSite = Boolean(state.sites.active);
+        return [
+          messageSharedWorker(refreshAuthToken()),
+          ...(hasActiveSite
+            ? [messageSharedWorker(openSiteSocket({ site: state.sites.active, xsrfToken: getXSRFToken() }))]
```

Comment:
Right now it opening the socket connection on login complete (not only re-login). Should it be only when logging in after a session expiration?